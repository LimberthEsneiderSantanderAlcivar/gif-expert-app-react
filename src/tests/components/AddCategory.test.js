import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"

describe('preubas en el componebnete <AddCategoy/> ', () => {

    const setcategories =jest.fn();
    let wrapper = shallow(<AddCategory setcategories={setcategories}/>)
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setcategories={setcategories}/>)

    });
    
    test('debe mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target:{value} });
        expect(wrapper.find('p').text().trim()).toBe(value);
    })
    
    
    test('no debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault:()=>{} });
        
        
        expect(setcategories).not.toHaveBeenCalled();
    })
    
    test('debe de llamar el setcategories y limpiuar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target:{value} });
        wrapper.find('form').simulate('submit', {preventDefault:()=>{} });
        expect(setcategories).toHaveBeenCalled();
        expect(wrapper.find('p').text().trim()).toBe('');
        
    })
    
    
})
