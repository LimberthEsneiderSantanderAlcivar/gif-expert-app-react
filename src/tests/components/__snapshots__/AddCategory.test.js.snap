// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preubas en el componebnete <AddCategoy/>  debe mostrar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
